Code/Framework/AzFramework/Tests/Platform/Common/Xcb/MockXcbInterface.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include <xcb/xcb.h>
+
+#define explicit ExplicitIsACXXKeyword
+#include <xcb/xkb.h>
+#undef explicit
+#include <xkbcommon/xkbcommon.h>
+
+#include "Printers.h"
+
+// xcb / xkb do not provide definitions of these structs, but the tests need some concrete value for them.
+struct xcb_connection_t
+{
+};
+
+struct xkb_context
+{
+};
+
+struct xkb_keymap
+{
+};
+
+struct xkb_state
+{
+};
+
+class MockXcbInterface
+{
+public:
+    MockXcbInterface()
+    {
+        self = this;
+    }
+    MockXcbInterface(const MockXcbInterface&) = delete;
+    MockXcbInterface(MockXcbInterface&&) = delete;
+    MockXcbInterface& operator=(const MockXcbInterface&) = delete;
+    MockXcbInterface& operator=(MockXcbInterface&&) = delete;
```

Comment:
Explicitly deleting the copy constructor, would implicitly delete the move constructor(because the copy constructor has been declared)

Reply:
I tend to go by the [rule of 5](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c21-if-you-define-or-delete-any-copy-move-or-destructor-function-define-or-delete-them-all), if you define or delete any of the special member functions, define or delete them all.